### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by jsoref**

### CODE REVIEW

**Comment by jsoref** [comment]
@jglick @daniel-beck can I get this merged?

File: src/test/java/hudson/slaves/CommandConnectorTest.java [hunk 1]
```diff
@@ -23,14 +23,18 @@
...
+import org.junit.Rule;
 
 /**
  * @author Kohsuke Kawaguchi
  */
-public class CommandConnectorTest extends HudsonTestCase {
-    public void testConfigRoundtrip() throws Exception {
+public class CommandConnectorTest {
+    @Rule public JenkinsRule j = new JenkinsRule();
+    @Test public void testConfigRoundtrip() throws Exception {
```
Comment (jglick) [inline src/test/java/hudson/slaves/CommandConnectorTest.java]: Note that the `test` prefix may be omitted in method names.

File: src/test/java/hudson/slaves/CommandLauncherTest.java [hunk 2]
```diff
@@ -71,18 +73,17 @@ public void commandSucceedsWithoutChannel() throws Exception {
...
-                    "dummy",
-                    j.createTmpDir().getPath(),
-                    "1",
-                    Node.Mode.NORMAL,
-                    "",
-                    new CommandLauncher(command),
-                    RetentionStrategy.NOOP,
-                    Collections.EMPTY_LIST
-            );
+            TemporaryFolder temporaryFolder = new TemporaryFolder();
```
Comment (jglick) [inline src/test/java/hudson/slaves/CommandLauncherTest.java]: This should be a `@Rule`.

File: src/test/java/hudson/slaves/CommandLauncherTest.java [hunk 3]
```diff
@@ -71,18 +73,17 @@ public void commandSucceedsWithoutChannel() throws Exception {
...
+            temporaryFolder.create();
+            File tempDir = temporaryFolder.newFolder();
+
+            slave = new DumbSlave("dummy", tempDir.getAbsolutePath(), new CommandLauncher(command));
+            slave.setNodeDescription("dummy");
+            slave.setNumExecutors(1);
+            slave.setMode(Node.Mode.NORMAL);
+            slave.setRetentionStrategy(RetentionStrategy.NOOP);
```
Comment (jglick) [inline src/test/java/hudson/slaves/CommandLauncherTest.java]: Some if not all of these setters can be omitted.

File: src/test/java/hudson/slaves/CommandLauncherTest.java (hunk 2 above)
Comment (jsoref) [reply src/test/java/hudson/slaves/CommandLauncherTest.java]: #6 is already introducing that:
https://github.com/jsoref/command-launcher-plugin/blob/6ec1d885e0ca9a91316e485993a222e96f489a9f/src/test/java/hudson/slaves/CommandLauncherTest.java#L109

These two PRs conflict a bit. If you have a preferred order...

File: src/test/java/hudson/slaves/CommandLauncherTest.java (hunk 3 above)
Comment (jsoref) [reply src/test/java/hudson/slaves/CommandLauncherTest.java]: The default num executors is `2`, I was just trying to expand the original arguments to their underlying implementation, but the test doesn't care, so I'm happy to remove them.

**Pull request merged**